core/domain/stats_domain.py

```diff
@@ -1067,7 +1067,7 @@ def validate(self):
                 'Expected answer_group_index to be an integer, received %s' %
                 str(self.answer_group_index))
 
-        if not isinstance(self.rule_spec_index, int):
+        if self.rule_spec_index and not isinstance(self.rule_spec_index, int):
```

Comment:
add tests?

Reply:
OK, I think I'm not understanding the point of this.

Firstly, what happens if self.rule_spec_index is 0 or the empty string? That wouldn't really get caught here.

Secondly, if the aim is to check for "not None", then probably should say "if X is not None"

Reply:
Oh, I didn't realize that. Thanks.

Reply:
Done. Fixed.